package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+      // "" or undefined -> null conversion
+
+      _styleClass: null,
+
+      /**
+       * Gets or sets the CSS class associated with this type.
+       *
+       * This attribute is typically used to associate an icon with a type.
+       *
+       * Attempting to set to a non-string value type implicitly
+       * converts the value to a string before assignment.
+       *
+       * An empty string or `undefined` value is interpreted as `null`.
+       *
+       * @type {?nonEmptyString}
+       */
+      get styleClass() {
+        return this._styleClass;
+      },
+
+      set styleClass(value) {
+        // undefined or "" -> null conversion
+        this._styleClass = nonEmptyString(value);
+      },
+
+      // TODO: implement inheritedStyleClasses
+      /**
+       * Gets the style classes of this and any base types.
+       *
+       * @type string[]
+       * @readonly
+       */
+      get inheritedStyleClasses() {
+        /* istanbul ignore next: not implemented yet */
+        throw error.notImplemented("Implement me!");
+      },
+      //endregion
+
+      //region ordinal property
+      // @type integer
+      // -> Optional(0), Inherited, Configurable
+      _ordinal: 0,
+
+      /**
+       * Gets or sets the ordinal associated with this type.
+       *
+       * The ordinal is used to disambiguate the order with which a type (or an instance of it)
+       * is shown in a user interface.
+       *
+       * Setting to a {@link Nully} value causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `0`.
+       *
+       * @type {number}
+       * @see pentaho.type.Type#isBrowsable
+       * @see pentaho.type.Type#category
+       */
+      get ordinal() {
+        return this._ordinal;
+      },
+
+      set ordinal(value) {
+        if(value == null) {
+          this._resetOrdinal();
+        } else {
+          this._ordinal = Math.floor((+value) || 0);
+        }
+      },
+
+      _resetOrdinal: function() {
+        if(this !== _type) {
+          delete this._ordinal;
+        }
+      },
+      //endregion
+
+      //region view property
+
+      // TODO: When type is anonymous, is view an absolute path?
+
+      // -> nonEmptyString, Optional, Inherited, Configurable, Localized
+      // undefined -> inherit
+      // null -> clear
+      // "" -> null conversion
+
+      _view: null, // {value: any, promise: Promise.<Class.<View>>}
+
+     /**
+      * Gets or sets the default view for instances of this type.
+      *
+      * When a string,
+      * it is the id of the view's AMD module.
+      * If the string starts with `/` or `xyz:`, or ends with `.js`,
+      * the id is considered to be absolute,
+      * otherwise,
+      * it is relative to this type's id folder, and converted to an absolute id.
+      *
+      * Setting to `undefined` causes the view to be inherited from the ancestor type,
```

Comment:
Suggest:

Setting this to `undefined` causes the view to be inherited from the ancestor type,